src/plugins/intel_cpu/src/nodes/multinomial.cpp

```diff
@@ -0,0 +1,185 @@
+// Copyright (C) 2018-2023 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#include "multinomial.hpp"
+
+#include <openvino/op/multinomial.hpp>
+
+#include "ie_ngraph_utils.hpp"
+#include "shape_inference/custom/multinomial.hpp"
+
+namespace ov {
+namespace intel_cpu {
+namespace node {
+
+Multinomial::Multinomial(const std::shared_ptr<ov::Node>& op, const GraphContext::CPtr& context)
+    : Node(op, context, MultinomialShapeInferFactory(op)) {
+    std::string errorMessage;
+    if (!isSupportedOperation(op, errorMessage)) {
+        IE_THROW(NotImplemented) << (errorMessage);
+    }
+
+    auto multinomial_op = as_type_ptr<op::v13::Multinomial>(op);
+    m_with_replacement = multinomial_op->get_with_replacement();
+    m_global_seed = multinomial_op->get_global_seed();
+    m_log_probs = multinomial_op->get_log_probs();
+    m_op_seed = multinomial_op->get_op_seed();
+    m_output_precision = InferenceEngine::details::convertPrecision(multinomial_op->get_convert_type());
+
+    for (size_t i = 0lu; i < op->get_input_size(); i++) {
+        if (is_type<op::v0::Constant>(op->get_input_node_ptr(i))) {
+            m_const_inputs[i] = true;
+        }
+    }
+}
+
+bool Multinomial::isSupportedOperation(const std::shared_ptr<const ov::Node>& op, std::string& errorMessage) noexcept {
+    try {
+        if (op->get_type_info() != op::v13::Multinomial::get_type_info_static()) {
+            errorMessage = "Only Multinomial operation from the opset13 is supported by the CPU plugin.";
+            return false;
+        }
+    } catch (...) {
+        return false;
+    }
+    return true;
+}
+
+void Multinomial::getSupportedDescriptors() {
+    if (getParentEdges().size() != 2) {
+        THROW_CPU_NODE_ERR("has incorrect number of input edges.");
+    }
+    if (getChildEdges().size() != 1) {
+        THROW_CPU_NODE_ERR("has incorrect number of output edges.");
+    }
+}
+
+void Multinomial::initSupportedPrimitiveDescriptors() {
+    m_probs_precision = getOriginalInputPrecisionAtPort(PROBS_PORT);
+    if (!one_of(m_probs_precision,
+                InferenceEngine::Precision::FP32,
+                InferenceEngine::Precision::FP16,
+                InferenceEngine::Precision::BF16)) {
+        m_probs_precision = InferenceEngine::Precision::FP32;
+    }
+
+    m_num_samples_precision = getOriginalInputPrecisionAtPort(NUM_SAMPLES_PORT);
+    if (!one_of(m_num_samples_precision, InferenceEngine::Precision::I64, InferenceEngine::Precision::I32)) {
+        m_num_samples_precision = InferenceEngine::Precision::I32;
+    }
+
+    if (!one_of(m_output_precision, InferenceEngine::Precision::I32)) {
```

Comment:
Is it required to check?
If precision is not `I32` then set to `I32`.  As is only one precision maybe set it fixed (const) and do not set or change/check?

Reply:
Good idea!